includes/import/class-wc-product-csv-importer.php

```diff
@@ -0,0 +1,277 @@
+		if ( false !== ( $handle = fopen( $this->file, 'r' ) ) ) {
+			$this->raw_keys = fgetcsv( $handle, 0, $this->params['delimiter'] );
+
+			if ( 0 !== $this->params['start_pos'] ) {
+				fseek( $handle, (int) $this->params['start_pos'] );
+			}
+
+			while ( false !== ( $row = fgetcsv( $handle, 0, $this->params['delimiter'] ) ) ) {
+				$this->raw_data[] = $row;
+
+	            if ( ( $this->params['end_pos'] > 0 && ftell( $handle ) >= $this->params['end_pos'] ) || 0 === --$this->params['lines'] ) {
+	            	break;
+				}
+			}
+		}
+
+		if ( ! empty( $this->params['mapping'] ) ) {
+			$this->set_mapped_keys();
+		}
+
+		if ( $this->params['parse'] ) {
+			$this->set_parsed_data();
+		}
+	}
+
+	/**
+	 * Set file mapped keys.
+	 *
+	 * @return array
+	 */
+	protected function set_mapped_keys() {
+		$mapping = $this->params['mapping'];
+
+		foreach ( $this->raw_keys as $key ) {
+			$this->mapped_keys[] = isset( $mapping[ $key ] ) ? $mapping[ $key ] : $key;
+		}
+	}
+
+	/**
+	 * Parse a comma-delineated field from a CSV.
+	 *
+	 * @param string $field
+	 * @return array
+	 */
+	protected function parse_comma_field( $field ) {
+		if ( empty( $field ) ) {
+			return array();
+		}
+
+		return array_map( 'esc_attr', array_map( 'trim', explode( ',', $field ) ) );
+	}
+
+	/**
+	 * Parse a field that is generally '1' or '0' but can be something else.
+	 *
+	 * @param string $field
+	 * @return bool|string
+	 */
+	protected function parse_bool_field( $field ) {
+		if ( '0' === $field ) {
+			return false;
+		}
+
+		if ( '1' === $field ) {
+			return true;
+		}
+
+		// Don't return explicit true or false for empty fields or values like 'notify'.
+		return esc_attr( $field );
+	}
+
+	/**
+	 * Parse a float value field.
+	 *
+	 * @param string $field
+	 * @return float|string
+	 */
+	protected function parse_float_field( $field ) {
+		if ( '' === $field ) {
+			return $field;
+		}
+
+		return floatval( $field );
+	}
+
+	/**
+	 * Parse a category field from a CSV.
+	 * Categories are separated by commas and subcategories are "parent > subcategory".
+	 *
+	 * @param string $field
+	 * @return array of arrays with "parent" and "name" keys.
+	 */
+	protected function parse_categories( $field ) {
+		if ( empty( $field ) ) {
+			return array();
+		}
+
+		$sections = array_map( 'trim', explode( ',', $field ) );
+		$categories = array();
+
+		foreach ( $sections as $section ) {
+
+			// Top level category.
+			if ( false === strpos( $section, '>' ) ) {
+				$categories[] = array(
+					'parent' => false,
+					'name'   => esc_attr( $section ),
+				);
+
+			// Subcategory.
+			} else {
+				$chunks = array_map( 'trim', explode( '>', $section ) );
+				$categories[] = array(
+					'parent' => esc_attr( reset( $chunks ) ),
+					'name'   => esc_attr( end( $chunks ) ),
+				);
+			}
+		}
+
+		return $categories;
+	}
+
+	/**
+	 * Map and format raw data to known fields.
+	 *
+	 * @return array
+	 */
+	protected function set_parsed_data() {
```

Comment:
We're going to have to update this to better work with the data the importer expects:
1. Put all of the `meta:`-prefixed keys into one `meta_data ` array when parsing.
2. Group all the `Attribute * Values`, `Attribute * Visible `, and `Download * URL` into their own `attributes` and `downloads` arrays.

Currently, I don't think the parser will generate the data that the importer is expecting for those fields. I can do this as a separate PR.